convert2rhel/pkgmanager/__init__.py

```diff
@@ -63,3 +69,27 @@ def create_transaction_handler():
         from convert2rhel.pkgmanager.handlers.dnf import DnfTransactionHandler
 
         return DnfTransactionHandler()
+
+
+def clean_yum_metadata():
+    """Remove cached metadata from yum.
+
+    This is to make sure that Convert2RHEL works with up-to-date data from repositories before, for instance, querying
+    whether the system has the latest package versions installed, or before checking whether enabled repositories have
+    accessible URLs.
+    """
+    # We are using run_subprocess here as an alternative to call_yum_cmd
+    # which doesn't apply the correct --enablerepos option because at this point
+    # the tool didn't initialize the necessary functions in SystemInfo.
+    # The viable solution was calling the yum command as a subprocess manually
+    # instead of using that function wrapper.
```

Comment:
The wording in the comment seems odd. I can imagine that the reason we call run_subprocess here is that because at the point in time we are calling this function in `main`, we don't have the `SystemInfo` initialized yet, but, if for some reason I decide to call this function just before validating the main transaction, then `SystemInfo` would be already initialized, and then the constraint here would not be valid anymore. 

```suggestion
    # We are using run_subprocess here as an alternative to call_yum_cmd
    # which doesn't apply the correct --enablerepos option because, if we call this
    # earlier in the conversion the tool didn't initialize the necessary functions in SystemInfo.
    # The viable solution was calling the yum command as a subprocess manually
    # instead of using that function wrapper.
```

What do you think?

Reply:
That's a good point, I'll add this and change "didn't" to "doesn't"